sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptions.java

```diff
@@ -413,6 +414,40 @@ public Long create(PipelineOptions options) {
 
   void setUserAgent(String userAgent);
 
+  /**
+   * A string defining whether GroupByKey transforms should be replaced by GroupByEncryptedKey
+   *
+   * <p>Beam will infer the secret type and value based on the secret itself. This guarantees that
+   * any data at rest during the performing a GBK, so this can be used to guarantee that data is not
+   * unencrypted. Runners with this behavior include the Dataflow, Flink, and Spark runners. The
+   * option should be structured like:
+   *
+   * <pre><code>
+   * --gbek=type:<secret_type>;<secret_param>:<value>
+   * </code></pre>
+   *
+   * for example:
+   *
+   * <pre><code>
+   * --gbek=type:GcpSecret;version_name:my_secret/versions/latest
+   * </code></pre>
+   *
+   * All variables should use snake case to allow consistency across languages.
```

Comment:
iirc "GBEK" will convert to "g_b_e_k" in Python pipeline. (e.g. in Java `--tempLocation` corresponds to python `--temp_location`)

Reply:
I noticed this a minute ago writing a xlang test - updated to avoid this, thanks